db/compaction/compaction_picker_fifo.cc

```diff
@@ -284,15 +285,18 @@ Compaction* FIFOCompactionPicker::PickSizeCompaction(
   return c;
 }
 
-Compaction* FIFOCompactionPicker::PickCompactionToWarm(
+Compaction* FIFOCompactionPicker::PickTemperatureChangeCompaction(
     const std::string& cf_name, const MutableCFOptions& mutable_cf_options,
     const MutableDBOptions& mutable_db_options, VersionStorageInfo* vstorage,
     LogBuffer* log_buffer) {
-  if (mutable_cf_options.compaction_options_fifo.age_for_warm == 0) {
+  const std::vector<FileTemperatureAge>& ages =
+      mutable_cf_options.compaction_options_fifo
+          .file_temperature_age_thresholds;
+  if (ages.empty()) {
     return nullptr;
   }
 
-  // PickCompactionToWarm is only triggered if there is no non-L0 files.
+  // Does not apply to multi-level FIFO.
```

Comment:
Can Open() return `Status::NotSupported()` if `file_temperature_age_thresholds` is nonempty and `num_levels > 1`?

I see this is dynamic in that it only bails out if the L1+ levels actually contain data. Is that something we can figure out at Open() time based on the config and state of the DB?

Reply:
Yes, added relevant checks in `ColumnFamilyData::ValidateOptions()` so that NotSupported() is returned during DB open and SetOptions(). The check in compaction picker is also updated to look at num_levels.